core/src/main/java/org/apache/iceberg/ManifestLists.java

```diff
@@ -66,6 +66,9 @@ static ManifestListWriter write(
       case 2:
         return new ManifestListWriter.V2Writer(
             manifestListFile, snapshotId, parentSnapshotId, sequenceNumber);
+      case 3:
+        return new ManifestListWriter.V3Writer(
+            manifestListFile, snapshotId, parentSnapshotId, sequenceNumber);
```

Comment:
Something I've been thinking about is abstracting away the different fields required for V1/V2/V3 without having to have a separate writer implementation for each. But it may not be worth it since the number of table format versions doesn't change too much and that sort of refactoring can happen down the line if it's determined to be needed (since these classes are package private, not in the public API I think there's that flexibility)

Reply:
Yeah I was figuring for now we keep things the same as between V1 and V2, also protects us a little more from inadvertently breaking V2 as we begin mucking about with V3.